File: Primes.java
import java.util.*;

public class Primes {
   public static void main(String[] args) {
      ArrayList<Integer> answer = Primes.primes_calculator(1000000);
      System.out.println(answer.get(10000));
   }

   public static ArrayList<Integer>primes_calculator(Integer num) {//ArrayList<Integer> primes_calculator(Integer num) {
      //Integer[] array;
      ArrayList<Integer> array = new ArrayList<Integer>();
      for (Integer i=2; i<=num; i++) {
         array.add(i);
      }
      HashSet<Integer> candidates = new HashSet<Integer>();
      for (Integer i=0; i<array.size(); i++) {
         candidates.add(array.get(i));
      }
      for (Integer i=0; i<array.size(); i++) {
         if (i%1000==0) {System.out.println(i);}
         Integer cur = array.get(i);
         if (candidates.contains(cur)) {
            for (Integer j=2; j<num/cur+1; j++) {
               if (candidates.contains(j*cur)) {candidates.remove(j*cur);}
            }
         }
      }
      ArrayList<Integer> arrayout;
      //arrayout = new Integer[candidates.size()];
      //for (Integer i=0; arrayout.size(); i++)
         //arrayout[i] = candidates[i];
      arrayout = new ArrayList<Integer>(candidates);
      Collections.sort(arrayout);
      return arrayout;
   }
}

File: primes.py
import math

def primes(n):
    candidates = set(range(2, n))

    for i, num in enumerate(list(candidates)[:len(candidates)/2]):
        for candidate in list(candidates)[i+1:]:
            if candidate % num == 0:
                candidates.remove(candidate)

    return list(candidates)

def largest_prime_factor(n, candidates, minimum):
    for candidate in candidates:
        if candidate <= minimum:
            next

        if n % candidate == 0:
            return n / candidate

    return None


def primes_alt(n):
    candidates = set(xrange(2, n))

    for num in list(candidates)[:len(candidates)/2+1]: #[:len(candidates)/2]:
        if num not in candidates:
           next
        for j in xrange(2,n/num+1):
           if j*num in candidates:
              #print "num,j"
              #print num
              #print j
              candidates.remove(j*num)

    return list(candidates)
